go/flow/catalog.go

```diff
@@ -106,6 +106,21 @@ func NewCatalog(ctx context.Context, etcd *clientv3.Client, root string) (Catalo
 	return catalog, nil
 }
 
+// GetTaskWithRevision returns the named CatalogTask, Commons, and its Commons ModRevision. It will
+// first wait for the named task to be at the given revision.
+func (c Catalog) GetTaskWithRevision(ctx context.Context, name string, minRevision int64) (*pf.CatalogTask, *Commons, int64, error) {
+	c.Mu.RLock()
+	defer c.Mu.RUnlock()
+	var task, commons, revision, err = c.getTask(name)
+	if revision < minRevision {
```

Comment:
Just WaitForRevision prior to reading out the task to remove a code path. It's super cheap if you're already >= the revision.

Also, can this be collapsed into `GetTask` ? Its callers, with the exception of ingest, need to pass in create revision to be fully correct w.r.t. data races, so I like that it makes that ordering constraint explicit.  Places where we really don't care (ingest) can pass zero.